src/main/java/gtPlusPlus/xmod/gregtech/common/covers/CoverOverflowValve.java

```diff
@@ -0,0 +1,373 @@
+package gtPlusPlus.xmod.gregtech.common.covers;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.nbt.NBTBase;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraftforge.common.util.ForgeDirection;
+import net.minecraftforge.fluids.Fluid;
+import net.minecraftforge.fluids.FluidStack;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import com.google.common.io.ByteArrayDataInput;
+import com.gtnewhorizons.modularui.api.screen.ModularWindow;
+import com.gtnewhorizons.modularui.common.widget.TextWidget;
+
+import gregtech.api.gui.modularui.CoverUIBuildContext;
+import gregtech.api.gui.modularui.GTUITextures;
+import gregtech.api.interfaces.metatileentity.IMetaTileEntity;
+import gregtech.api.interfaces.tileentity.ICoverable;
+import gregtech.api.metatileentity.CommonMetaTileEntity;
+import gregtech.api.metatileentity.implementations.MTEBasicTank;
+import gregtech.api.metatileentity.implementations.MTEFluid;
+import gregtech.api.util.CoverBehaviorBase;
+import gregtech.api.util.GTUtility;
+import gregtech.api.util.ISerializableObject;
+import gregtech.common.gui.modularui.widget.CoverDataControllerWidget;
+import gregtech.common.gui.modularui.widget.CoverDataFollowerNumericWidget;
+import gregtech.common.gui.modularui.widget.CoverDataFollowerToggleButtonWidget;
+import io.netty.buffer.ByteBuf;
+
+public class CoverOverflowValve extends CoverBehaviorBase<CoverOverflowValve.OverflowValveData> {
+
+    private final int minOverflowPoint = 0;
+    private final int maxOverflowPoint;
+
+    public CoverOverflowValve(int maxOverflowPoint) {
+        super(OverflowValveData.class);
+        this.maxOverflowPoint = maxOverflowPoint;
+    }
+
+    @Override
+    public OverflowValveData createDataObject() {
+        return new OverflowValveData(0, 0, false, false);
+    }
+
+    @Override
+    public OverflowValveData createDataObject(int aLegacyData) {
+        return new OverflowValveData(0, 0, false, false);
+    }
+
+    private FluidStack doOverflowThing(FluidStack fluid, OverflowValveData data) {
+        if (fluid != null && fluid.amount > data.overflowPoint)
+            fluid.amount -= Math.min(fluid.amount - data.voidingRate, data.voidingRate);
```

Comment:
Current logic *dupes* fluid if voiding rate is greater than overflow point. It also doesn't stop voiding when reaches overflow point. 
```suggestion
            fluid.amount = Math.min(fluid.amount - data.voidingRate, data.overflowPoint);
```

Reply:
> Current logic _dupes_ fluid if voiding rate is greater than overflow point. It also doesn't stop voiding when reaches overflow point.

The voiding rate can't be higher than `maxOverflowPoint` by the logic I written in widget or it's happens ?

Reply:
> Current logic _dupes_ fluid if voiding rate is greater than overflow point. It also doesn't stop voiding when reaches overflow point.
 
During refactoring I done a mess... as usual

Reply:
1) It's totally possible to set ***current*** voiding rate larger than overflow value (while you're speaking about maximum limit):
![image](https://github.com/user-attachments/assets/144527c3-a7d6-4d45-b0e6-6557262ef0a4)
2) I made mistake in my code sample. Of course there should be `max` function.
3) Your logic causes dupe:
* Given: amount is 2 000, voiding rate is 10 000 and overflow value is 1 000.
* `amount > overflowValue` - true.
* `Math.min(amount - voidingRate, amount)` = `Math.min(2000 - 10000, 2000)` = -8000.
* `amount -= Math.min(...)`: `amount -= -8000`. *Oops,  8 000 L of liquid is duped*.

So please use suggested code (with coreection: `max` instead of `min`).

Reply:
1) It's totally possible to set ***current*** voiding rate higher than overflow point.
![image](https://github.com/user-attachments/assets/15ee1cb4-45b0-4509-bf6b-b08679cd7fff)
2) I made a typo in suggested code. Of course there should be `max` instead of `min`.
3) How your current logic duplicates fluids:
* Given container with 1000 L of liquid, cover is set like shown on screen.
* Adding any amount of liquid causes condition `amount > overflowValue` to be true (1000 + 1 > 1000).
* `Math.min(amount - voidingRate, voidingRate)` = `Math.min(1001 - 10000, 10000)` = -8999
* `amount -= Math.min(...)` = `amount -= -8999`. *Oops, over 8000 L of liquid is duped!*

So please use suggested code (with exception: `max` instead of `min`).

Reply:
I have a lot of thing to learn :face_with_head_bandage:

Reply:
Done in commit https://github.com/GTNewHorizons/GT5-Unofficial/pull/3040/commits/3c1274c905452cea324eb5e411f1e99ead798933